python-package/xgboost/config.py

```diff
@@ -0,0 +1,101 @@
+"""Global configuration for XGBoost"""
+import ctypes
+from contextlib import contextmanager
+
+from .core import _LIB, _check_call
+
+
+def set_config(*, verbosity=None):
```

Comment:
Can we avoid listing parameter and use opaque strings similar to `booster.set_param`.

Reply:
It's a bit nicer to explicitly parameters, so that the Python API doc will show it. Otherwise, users won't know which parameters are in the global scope. See https://scikit-learn.org/stable/modules/generated/sklearn.config_context.html

Reply:
Another option is to list global parameters in https://xgboost.readthedocs.io/en/latest/parameter.html.

Reply:
I'm going to honor the spirit of avoiding duplication, and will avoid listing parameter in the Python later. Instead, we'll have a documented list in `parameter.rst`.

Reply:
See the latest commit, which uses opaque strings like `set_param`.